hoomd/hpmc/PairPotentialLennardJones.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2009-2023 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#include "PairPotentialLennardJones.h"
+
+namespace hoomd
+    {
+namespace hpmc
+    {
+
+PairPotentialLennardJones::PairPotentialLennardJones(std::shared_ptr<SystemDefinition> sysdef)
+    : PairPotential(sysdef), m_params(m_type_param_index.getNumElements())
+    {
+    }
+
+LongReal PairPotentialLennardJones::energy(const LongReal r_squared,
+                                           const vec3<LongReal>& r_ij,
+                                           const unsigned int type_i,
+                                           const quat<LongReal>& q_i,
+                                           const LongReal charge_i,
+                                           const unsigned int type_j,
+                                           const quat<LongReal>& q_j,
+                                           const LongReal charge_j) const
+    {
+    unsigned int param_index = m_type_param_index(type_i, type_j);
+    const auto& param = m_params[param_index];
+
+    LongReal lj1 = param.epsilon_x_4 * param.sigma_6 * param.sigma_6;
+    LongReal lj2 = param.epsilon_x_4 * param.sigma_6;
```

Comment:
The compiler is likely smart enough for this, but just in case.
```suggestion
    LongReal lj2 = param.epsilon_x_4 * param.sigma_6;
    LongReal lj1 = lj2 * param.sigma_6;
```